core/ssh_keys_test.go

```diff
@@ -0,0 +1,476 @@
+	tc := testclient.New(nil, nil, nil)
+	c := NewSSHKeysClient(tc)
+	assert.Equal(t, tc, c.client)
+	assert.Equal(t, &url.URL{Path: "/core/v1/"}, c.basePath)
+}
+
+func TestSSHKeysClient_List(t *testing.T) {
+	type args struct {
+		ctx  context.Context
+		org  OrganizationRef
+		opts *ListOptions
+	}
+	tests := []struct {
+		name     string
+		args     args
+		resp     *katapult.Response
+		respErr  error
+		respV    *sshKeysResponseBody
+		want     []*AuthSSHKey
+		wantReq  *katapult.Request
+		wantResp *katapult.Response
+		wantErr  string
+	}{
+		{
+			name: "success",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+				opts: &ListOptions{
+					Page:    5,
+					PerPage: 32,
+				},
+			},
+			resp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+			respV: &sshKeysResponseBody{
+				Pagination: &katapult.Pagination{Total: 394},
+				SSHKeys: []*AuthSSHKey{
+					{ID: "ssh_O574YEEEYeLmqdmn"},
+				},
+			},
+			want: []*AuthSSHKey{
+				{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"page":     []string{"5"},
+						"per_page": []string{"32"},
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+			wantResp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 394},
+				Response:   &http.Response{StatusCode: http.StatusOK},
+			},
+		},
+		{
+			name: "success with nil options",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			resp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+			respV: &sshKeysResponseBody{
+				Pagination: &katapult.Pagination{Total: 333},
+				SSHKeys: []*AuthSSHKey{
+					{ID: "ssh_O574YEEEYeLmqdmn"},
+				},
+			},
+			want: []*AuthSSHKey{
+				{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+			wantResp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 333},
+				Response:   &http.Response{StatusCode: http.StatusOK},
+			},
+		},
+		{
+			name: "request error",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			respErr: fmt.Errorf("flux capacitor undercharged"),
+			wantErr: "flux capacitor undercharged",
+		},
+		{
+			name: "request error with nil response",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			resp:    nil,
+			respErr: fmt.Errorf("something is really wrong"),
+			wantErr: "something is really wrong",
+		},
```

Comment:
It looks like these two test cases are effectively identical. Both end up with a nil `resp` value. To ensure we can handle errors both with and without a response value from the client object, the test cases need to look something like this:

```suggestion
		{
			name: "request error",
			args: args{
				ctx: context.Background(),
				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
			},
			resp: &katapult.Response{
				Response: &http.Response{
					StatusCode: http.StatusInternalServerError,
				},
			},
			respErr: fmt.Errorf("flux capacitor undercharged"),
			wantResp: &katapult.Response{
				Response: &http.Response{
					StatusCode: http.StatusInternalServerError,
				},
			},
			wantErr: "flux capacitor undercharged",
		},
		{
			name: "request error with nil response",
			args: args{
				ctx: context.Background(),
				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
			},
			resp:    nil,
			respErr: fmt.Errorf("something is really wrong"),
			wantErr: "something is really wrong",
		},
```